### DISCUSSION TITLE
API Naming Conventions

### DISCUSSION THREAD

**Discussion by uklotzde** [Serial-ATA/lofty-rs#225] [Q&A]
How about following the [Rust API Guidelines for naming](https://rust-lang.github.io/api-guidelines/naming.html), e.g. use `FrameId` instead of `FrameID` or `AppleXid` instead of `AppleXID`? Consistency reduces cognitive load.

**Comment by Serial-ATA** [comment]
I wasn't aware of those guidelines.

I'm not opposed to following them, but many types had their capitalization changed just a few months ago in [0.7.0](https://github.com/Serial-ATA/lofty-rs/blob/main/CHANGELOG.md#070---2022-06-27). Could be annoying downstream to have to change those names *again*. But it's better to change them sooner than later I suppose.

**Comment by uklotzde** [comment]
In the past (before Rust) I was also using UPPERCASE literals in type names. But after getting used to the UpperCamelCase-style I found the latter more pleasant to read and recognize.

**Comment by uklotzde** [comment]
For many literals/abbreviations it is often unclear what the "official" casing actually is and sometimes it even changes with time. Simply following the convention of your programming language protects you from those external imponderables that are out of your control.

It is unfortunate that the naming has been changed in 0.7. But this is still a 0.x version and there is a good reason to revert this decision.
